// react-native-background-audio/Component.js
import React from 'react';
import Expo, { Audio } from 'expo-av';

export default class BgAudio extends React.Component {

  constructor(props) {
    super(props);
    this.source = props.options.source

  }
  // Initializes the audio on component mount.
  UNSAFE_componentWillMount() {
    if (this.source.local) {
      this.play(this.source.local)
  
    }
    else if (this.source.remote) {
      this.play(this.source.remote)
     
    }
  }
  // Plays the source passed in by the Props object
  play = (source) => {

    Audio.setIsEnabledAsync(true);
    Audio.setAudioModeAsync({
      playThroughEarpieceAndroid: true,
      interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
      interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX,
    });
    this.sound = new Audio.Sound();

    const play_sound = (async () => {

      await this.sound.loadAsync(source); // props path to asset 
      await this.sound.playAsync();

      
    })();
  }


  // Does not render any view 
  render() {
    return (
      null
    );
  }
}
